students/dcastrowa/lesson03/assignment/src/basic_operations.py

```diff
@@ -0,0 +1,204 @@
+            logger.info(f'id: {new_customer.customer_id}')
+            logger.info(f'first: {new_customer.first_name}')
+            logger.info(f'last: {new_customer.last_name}')
+            logger.info(f'address: {new_customer.home_address}')
+            logger.info(f'phone: {new_customer.phone_number}')
+            logger.info(f'email: {new_customer.email_address}')
+            logger.info(f'status: {new_customer.status}')
+            logger.info(f'limit: {new_customer.credit_limit}')
+    except Exception as e:
+        logger.warning(f'Error creating = ID: {customer_id}')
+        logger.warning(e)
+
+    return Customer
+
+
+def search_customer(customer_id):
+    """
+    search for customer by customer id
+    :return: (dict) customer data
+    """
+    with database.transaction():
+        query = (Customer
+                 .select(Customer)
+                 .where(Customer.customer_id == customer_id)
+                 )
+        logger.info(query)
+
+    logger.info('iterate through query to create dictionary')
+    results = {}
+    for item in query:
+        results['first_name'] = item.first_name
+        results['last_name'] = item.last_name
+        results['email'] = item.email_address
+        results['phone_number'] = item.phone_number
+
+    if not bool(results):
+        logger.warning('Customer not found')
+
+    return results
+
+
+def delete_customer(customer_id):
+    """
+    delete customer by customer id
+    :return:
+    """
+    try:
+        with database.transaction():
+            customer = Customer.get(Customer.customer_id == customer_id)
+
+            logger.info(f'Trying to delete {customer.first_name}')
+            customer.delete_instance()
+            logger.info(f'Deleted {customer.first_name}')
+
+            return True
+
+    except Exception as e:
+        logger.warning(f'Customer ID: {customer_id} does not exist')
+        logger.warning(e)
+
+        return False
+
+
+def update_customer(customer_id, credit_limit):
+    """
+    update customer credit limit by customer id
+    :return:
+    """
+    try:
+        with DATABASE.transaction():
+            customer_update = Customer.get(Customer.customer_id == customer_id)
+            logger.info(f'Current limit: {customer_update.credit_limit}')
+            customer_update.credit_limit = credit_limit
+            logger.info(f'New credit limit: {customer_update.credit_limit}')
+
+            return True
+
+    except Exception as e:
+        logger.warning(f'Customer ID: {customer_id} does not exist')
+        logger.warning(e)
+
+        return False
+
+
+def list_active_customers():
+    """
+    gets total number of active customers
+    :return: (int) # of active customers
+    """
+    with database.transaction():
+        query = (Customer
+                 .select(fn.COUNT(Customer.status).alias('count'))
+                 .where(Customer.status))
+        logger.info(query)
+
+    active_customers = []
```

Comment:
Could you do this differently? Hint: aggregate function (count); or, you could increase the counter within the loop